spring-cloud-stream-rxjava/src/main/java/org/springframework/cloud/stream/annotation/rxjava/SubjectMessageHandler.java

```diff
@@ -62,46 +62,80 @@
  * @author Ilayaperumal Gopinathan
  */
 @SuppressWarnings({"unchecked", "rawtypes"})
-public class SubjectMessageHandler extends AbstractMessageProducingHandler implements DisposableBean {
+public class SubjectMessageHandler extends AbstractMessageProducingHandler implements SmartLifecycle {
 
 	private final Logger logger = LoggerFactory.getLogger(getClass());
 
 	@SuppressWarnings("rawtypes")
 	private final RxJavaProcessor processor;
 
-	private final Subject subject;
+	private volatile Subject subject;
 
-	private final Subscription subscription;
+	private volatile Subscription subscription;
+
+	private volatile boolean running = false;
 
 	@SuppressWarnings({"unchecked", "rawtypes"})
 	public SubjectMessageHandler(RxJavaProcessor processor) {
 		Assert.notNull(processor, "RxJava processor must not be null.");
 		this.processor = processor;
-		subject = new SerializedSubject(PublishSubject.create());
-		Observable<?> outputStream = processor.process(subject);
-		subscription = outputStream.subscribe(new Action1<Object>() {
-			@Override
-			public void call(Object outputObject) {
-				if (ClassUtils.isAssignable(Message.class, outputObject.getClass())) {
-					getOutputChannel().send((Message) outputObject);
+	}
+
+	@Override
+	public synchronized void start() {
+		if (!running) {
+			subject = new SerializedSubject(PublishSubject.create());
+			Observable<?> outputStream = processor.process(subject);
+			subscription = outputStream.subscribe(new Action1<Object>() {
+
+				@Override
+				public void call(Object outputObject) {
+					if (ClassUtils.isAssignable(Message.class, outputObject.getClass())) {
+						getOutputChannel().send((Message) outputObject);
+					}
+					else {
+						getOutputChannel().send(MessageBuilder.withPayload(outputObject).build());
+					}
 				}
-				else {
-					getOutputChannel().send(MessageBuilder.withPayload(outputObject).build());
+			}, new Action1<Throwable>() {
+
+				@Override
+				public void call(Throwable throwable) {
+					logger.error(throwable.getMessage(), throwable);
+				}
+			}, new Action0() {
+
+				@Override
+				public void call() {
+					logger.error("Subscription close for [" + subscription + "]");
```

Comment:
Not from the changes on this PR but this logger could be at `info` level as this method is invoked `onCompleted` and we may not want to log it as `error`.